craft_parts/overlays/overlays.py

```diff
@@ -0,0 +1,125 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Overlay handling helpers."""
+
+import logging
+import os
+from pathlib import Path
+from typing import Set, Tuple
+
+logger = logging.getLogger(__name__)
+
+
+def visible_in_layer(srcdir: Path, destdir: Path) -> Tuple[Set[str], Set[str]]:
```

Comment:
src/dst seems a bit unusual in this context - maybe rewrite as lower/upper?
```suggestion
def visible_in_layer(lower_dir: Path, upper_dir: Path) -> Tuple[Set[str], Set[str]]:
```